pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepoManager.java

```diff
@@ -334,6 +371,178 @@ public RepositoryData generate() {
+                                Collections.emptyList(), // extra maven repos, ignore this
+                                "test",
+                                "provided" // dependency scopes that should be ignored
+                        );
+                    }
+                }
+            }
+            return repackage(sourceDir);
+        } catch (Exception bme) {
+            bme.printStackTrace();
+            throw new RuntimeException("Unable to resolve and package. " + bme.getLocalizedMessage());
+        }
+    }
+
+    public Map<Artifact, String> collectRedhatVersions(List<Artifact> extensionArtifacts) {
+        Map<Artifact, String> result = new HashMap<>();
+        builds.forEach((key, pncBuild) -> {
+            if (pncBuild.getBuiltArtifacts() != null) {
+                pncBuild.getBuiltArtifacts().forEach((artifactWrapper -> {
+                    extensionArtifacts.forEach((extensionArtifact -> {
+                        GAV gav = artifactWrapper.toGAV();
+                        if (extensionArtifact.getGroupId().equals(gav.getGroupId())
+                                && extensionArtifact.getArtifactId().equals(gav.getArtifactId())) {
+                            result.put(extensionArtifact, gav.getVersion());
+                        }
+                    }));
+                }));
+            }
+            if (pncBuild.getDependencyArtifacts() != null) {
+                pncBuild.getDependencyArtifacts().forEach((artifactWrapper -> {
+                    extensionArtifacts.forEach((extensionArtifact -> {
+                        GAV gav = artifactWrapper.toGAV();
+                        if (extensionArtifact.getGroupId().equals(gav.getGroupId())
+                                && extensionArtifact.getArtifactId().equals(gav.getArtifactId())) {
+                            result.put(extensionArtifact, gav.getVersion());
+                        }
+                    }));
+                }));
+            }
+        });
+
+        return result;
+    }
+
+    /**
+     * @param urlString url to extensionList file. File must contains the extensions in format of
+     *        groupId:artifactId:version:type:classifier classifier and type are optionals
+     * @return List of Artifacts
+     * @throws Exception
+     */
+    public List<Artifact> parseExtensionsArtifactList(String urlString) throws Exception {
+        try {
+
+            URL url = new URL(urlString);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            BufferedReader br = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+            ArrayList<Artifact> list = new ArrayList<>();
+            String buffer = "";
+            while (buffer != null) {
+                buffer = br.readLine();
+                if (buffer == null) {
+                    break;
+                }
+                String[] parts = buffer.split(":");
```

Comment:
`trim()` the buffer first, skip the line if it is empty, or starts with `#` maybe too?